pkg/command/context.go

```diff
@@ -384,6 +392,75 @@ func globalLogin(c *configtypes.Context) (err error) {
 	return nil
 }
 
+func selfManagedTMCLogin(c *configtypes.Context) (err error) {
+	issuer, err := getIssuerURLForTMCEndPoint(endpoint)
+	if err != nil {
+		return err
+	}
+	refreshToken := ""
+	token, err := csp.GetAccessTokenFromSelfManagedIDP(refreshToken, issuer)
+	if err != nil {
+		return err
+	}
+	if token == nil {
+		return errors.Errorf("token issuer %s returned nil token", issuer)
+	}
+
+	a := configtypes.GlobalServerAuth{}
+	a.Issuer = issuer
+	// TODO: parse claims from ID token to get org-tenant ID info
+	// a.UserName = claims.Username
+	// a.Permissions = claims.Permissions
+	a.AccessToken = token.AccessToken
+	a.IDToken = token.IDToken
+	a.RefreshToken = token.RefreshToken
+	a.Type = idTokenType
+	// time.Duration is a typedef for int64 and is in nanoseconds
+	expiresAt := (time.Now().Add(time.Duration(token.ExpiresIn)))
```

Comment:
The expiration is computed differently than for the non-self-managed case.  Is this on purpose?

In fact, `time.Now().Add()` seems to need a value in nano seconds, ~but `token.ExpiresIn` is in seconds~ (according to its doc).

Ok, so I see that `token.ExpiresIn` is in nanoseconds as set in `csp.GetAccessTokenFromSelfManagedIDP`.  This does not follow the API definition of that variable.  I think we should be consistent and set `token.ExpiresIn` to seconds in `csp.GetAccessTokenFromSelfManagedIDP`.

What do you think?

Reply:
sure, updated the `csp.GetAccessTokenFromSelfManagedIDP` and here to be inline with API definition. We might loose some precision, but I think that should be fine.